### ISSUE TITLE
easymotion scrolls to cursor if file was scrolled

### ISSUE THREAD

**Issue by fanantoxa** [VSCodeVim/Vim#4856]
**Describe the bug**
I have a cursor on line 2
I scroll file for few pages
When I activate `-- EASYMOTION INPUT --` (pressing `<Leader><Leader>s`
It scrolls me back to the very top where my cursor is.

But I wanted to navigate on the part of file where I was few screens down.

**To Reproduce**

1.  Open a file with many lines
2.  Set cursor on line 2
3.  Scroll down few screens (not move cursor, but scroll file)
4.  Activate `-- EASYMOTION INPUT --` (pressing `<Leader><Leader>s`
5. Get scrolled back to very top

**Expected behavior**
It doesn't scroll and I can navigate on exact scroll position

**Screenshots**
![easymotion](https://user-images.githubusercontent.com/7717033/81978511-48ad3c80-9634-11ea-90c4-7a478e0c76ff.gif)

**Environment (please complete the following information):**

- Extension (VsCodeVim) version: 1.14.2
- VSCode version: 1.45.0
- OS: Ubuntu 18.04

**Additional context**
If this is not considered as a bug is that any way how to make it stop do that or should I create feature request?

**Comment by berknam** [comment]
This is normal VSCode behavior. When you scroll with the mouse the cursor stays in same position even if it is off screen. In Vim this doesn't happen, when you scroll (even with the mouse) the cursor moves with the screen.

But in VSCodeVim if you scroll with `<C-e>`, `<C-y>` or `<C-d>`, `<C-u>` or `<C-f>`, `<C-b>` the cursor moves with the screen and then easymotion will start looking from where you are. Or after scrolling with the mouse to where you want you can click somewhere to move the cursor there.

We could try to implement the cursor moving along when scrolling with mouse, but I personally like this behavior. I some times scroll to quickly see something and then keep writing and the screen jumps back to my cursor. So if this gets implemented I think it would need an option to opt out.

**Comment by fanantoxa** [comment]
@berknam Hey!
> This is normal VSCode behavior. When you scroll with the mouse the cursor stays in same position even if it is off screen.
I understand this. And I do like . It really helpful when you want to just look around when you do some changes.
(That why I don't use vim actually, I don't what to move my cursor usually, just scroll around)

What I was thinking to have for example:
`<leader><leader> v <char>` (were `v` stand for `viewport`
So it will work the same as `<leader><leader> s <char>` But it won't scroll back to cursor and just show letter in current scroll position (aka viewport)

Not sure if it can be done but I saw that when you press `<leader><leader>` it doesn't scroll back to cursor and when you hit `s` and go into `-- EASYMOTION INPUT --` mode and scroll back to cursor

**Comment by laggardkernel** [comment]
> This is normal VSCode behavior. When you scroll with the mouse the cursor stays in same position even if it is off screen.

Yes. But in terminal vim, scrolling with mouse moves the cursor when the cursor reaches the top or the bottom of the viewport. In VSCode, scrolling with mouse only changes the viewport.

I think we can improve it by sticking with current view, but not jumping back the cursor position when vim-easymotion triggers. This is exactly what JetBrains/ideavim does.
